symengine/dense_matrix.cpp

```diff
@@ -1613,6 +1632,37 @@ void cross(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C)
 
 RCP<const Set> eigen_values(const DenseMatrix &A)
 {
+    unsigned n = A.nrows();
+    bool is_upper = true, is_lower = true;
+    for (unsigned i = 0; i < n; ++i) {
+        for (unsigned j = 0; j < n; ++j) {
+            if (i > j and not eq(*(A.get(i, j)), *zero)) {
+                is_lower = false;
+                break;
```

Comment:
You need to break both the loops. It's easier if this was a separate helper function.

Reply:
nvm

Reply:
So should we add a property is_upper and is_lower to Class DenseMatrix??